src/v2/System/Relay/SystemQueryRenderer.tsx

```diff
@@ -28,19 +29,26 @@ export function SystemQueryRenderer<T extends OperationType>({
   lazyLoad = false,
   lazyLoadThreshold = 1500,
   placeholder,
-  environment,
+  environment: _environment,
   variables = {},
   render,
   ...rest
 }: SystemQueryRendererProps<T>): JSX.Element {
   const isMounted = useDidMount()
+
   const { isEnteredView, Waypoint } = useLazyLoadComponent({
     threshold: lazyLoadThreshold,
   })
+
   const showPlaceholder =
     debugPlaceholder || !isMounted || (lazyLoad && !isEnteredView)
 
+  const { relayEnvironment } = useSystemContext()
+
+  const environment = _environment ?? relayEnvironment
+
   if (!environment) {
+    console.warn("Requires an environment")
```

Comment:
This took me way too long to track down 😅

Reply:
For system-level stuff can we use `v2/Utils/logger`